sql.go

```diff
@@ -83,27 +83,30 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+var nullJSON = []byte("null")
+
 // MarshalJSON marshals the NullUUID as null or the nested UUID
 func (u NullUUID) MarshalJSON() ([]byte, error) {
 	if !u.Valid {
-		return json.Marshal(nil)
+		return nullJSON, nil
 	}
-
-	return json.Marshal(u.UUID)
+	var buf [38]byte
+	buf[0] = '"'
+	encodeHex(buf[1:37], u.UUID)
+	buf[37] = '"'
+	return buf[:], nil
 }
 
 // UnmarshalJSON unmarshals a NullUUID
 func (u *NullUUID) UnmarshalJSON(b []byte) error {
-	if bytes.Equal(b, []byte("null")) {
+	if string(b) == "null" {
 		u.UUID, u.Valid = Nil, false
 		return nil
 	}
-
-	if err := json.Unmarshal(b, &u.UUID); err != nil {
-		return err
+	if n := len(b); n >= 2 && b[0] == '"' {
```

Comment:
Is this removing leading and trailing quotes?  If so, it's quite unintuitive to my eye.

`bytes.Trim(b, "\"")` is functionally equivalent and would be much less "clever" (as in wouldn't require an explanation)

Reply:
`bytes.Trim(b, "\"")` would make invalid  inputs that have either extraneous quotes or a quote on only one side valid since it would transform `""UUID""` to `UUID` and `"UUID` to `UUID`.